test/integration/resourcemanager/health/health_test.go

```diff
@@ -0,0 +1,339 @@
+					},
+				}}
+				Expect(testClient.Status().Patch(ctx, managedResource, patch)).To(Succeed())
+			})
+
+			It("sets ManagedResource to unhealthy as Pod is not ready", func() {
+				Eventually(func(g Gomega) []gardencorev1beta1.Condition {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+					return managedResource.Status.Conditions
+				}).Should(
+					containCondition(ofType(resourcesv1alpha1.ResourcesHealthy), withStatus(gardencorev1beta1.ConditionFalse), withReason("PodUnhealthy")),
+				)
+			})
+
+			It("sets ManagedResource to healthy as Pod is running", func() {
+				By("add resources to ManagedResource status")
+				patch := client.MergeFrom(pod.DeepCopy())
+				pod.Status.Phase = corev1.PodRunning
+				Expect(testClient.Status().Patch(ctx, pod, patch)).To(Succeed())
+
+				Eventually(func(g Gomega) []gardencorev1beta1.Condition {
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(managedResource), managedResource)).To(Succeed())
+					return managedResource.Status.Conditions
+				}).Should(
+					containCondition(ofType(resourcesv1alpha1.ResourcesHealthy), withStatus(gardencorev1beta1.ConditionTrue), withReason("ResourcesHealthy")),
+				)
+			})
+		})
+	})
+})
+
+func setCondition(managedResource *resourcesv1alpha1.ManagedResource, conditionType gardencorev1beta1.ConditionType, status gardencorev1beta1.ConditionStatus) {
+	managedResource.Status.Conditions = v1beta1helper.MergeConditions(managedResource.Status.Conditions, gardencorev1beta1.Condition{
+		Type:               conditionType,
+		Status:             status,
+		LastUpdateTime:     metav1.Now(),
+		LastTransitionTime: metav1.Now(),
+	})
+}
+
+func containCondition(matchers ...gomegatypes.GomegaMatcher) gomegatypes.GomegaMatcher {
+	return ContainElement(And(matchers...))
+}
+
+func ofType(conditionType gardencorev1beta1.ConditionType) gomegatypes.GomegaMatcher {
+	return gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+		"Type": Equal(conditionType),
+	})
+}
+
+func withStatus(status gardencorev1beta1.ConditionStatus) gomegatypes.GomegaMatcher {
+	return gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+		"Status": Equal(status),
+	})
+}
+
+func withReason(reason string) gomegatypes.GomegaMatcher {
+	return gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+		"Reason": Equal(reason),
+	})
+}
+
+func generatePodTestResource(name string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: testNamespace.Name,
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{{
+				Name:    "test",
+				Image:   "alpine",
+				Command: []string{"sh", "-c", "echo hello"},
+			}},
+			// set to non-existing node, so that no kubelet will interfere when testing against existing cluster, so that we
+			// solely control the pod's status
+			NodeName: "non-existing",
```

Comment:
This approach won't work for most resources, should we drop this to not suggest that it can be done similarly for other resources?

Reply:
> [...] to not suggest that it can be done similarly for other resources?

I'm not sure, what you mean. Can you elaborate?

For this test, I needed some way to produce an unhealthy objects and this was the easiest way, that also works on an existing cluster. I guess, for other resources you have to be creative as well 😄

Reply:
> I doubt that this test can be executed on an existing cluster [...] but this won't stay this way.

Hmm, why? I think, this test is very complete already, why shouldn't it stay this way? I'm confused 😄